packages/asset-buyer/src/quote_consumers/exchange_swap_quote_consumer.ts

```diff
@@ -0,0 +1,127 @@
+import { ContractWrappers, ContractWrappersError, ForwarderWrapperError, SignedOrder } from '@0x/contract-wrappers';
+import { AbiEncoder, abiUtils, BigNumber, providerUtils } from '@0x/utils';
+import { SupportedProvider, Web3Wrapper, ZeroExProvider } from '@0x/web3-wrapper';
+import { MethodAbi } from 'ethereum-types';
+import * as _ from 'lodash';
+
+import { constants } from '../constants';
+import {
+    CalldataInfo,
+    ExchangeMarketBuySmartContractParams,
+    SmartContractParamsInfo,
+    SwapQuote,
+    SwapQuoteConsumer,
+    SwapQuoteConsumerError,
+    SwapQuoteConsumerOpts,
+    SwapQuoteExecutionOpts,
+    SwapQuoteGetOutputOpts,
+} from '../types';
+import { assert } from '../utils/assert';
+import { swapQuoteConsumerUtils } from '../utils/swap_quote_consumer_utils';
+import { utils } from '../utils/utils';
+
+export class ExchangeSwapQuoteConsumer implements SwapQuoteConsumer<ExchangeMarketBuySmartContractParams> {
+
+    public readonly provider: ZeroExProvider;
+    public readonly networkId: number;
+
+    private readonly _contractWrappers: ContractWrappers;
+
+    constructor(supportedProvider: SupportedProvider, options: Partial<SwapQuoteConsumerOpts> = {}) {
+        const { networkId } = _.merge({}, constants.DEFAULT_SWAP_QUOTER_OPTS, options);
+        assert.isNumber('networkId', networkId);
+
+        const provider = providerUtils.standardizeOrThrow(supportedProvider);
+        this.provider = provider;
+        this.networkId = networkId;
+        this._contractWrappers = new ContractWrappers(this.provider, {
+            networkId,
+        });
+    }
+
+    public async getCalldataOrThrowAsync(quote: SwapQuote, opts: Partial<SwapQuoteGetOutputOpts>): Promise<CalldataInfo> {
+        assert.isValidSwapQuote('quote', quote);
+
+        const { params, to, ethAmount, methodAbi } = await this.getSmartContractParamsOrThrowAsync(quote, opts);
+        const abiEncoder = new AbiEncoder.Method(methodAbi);
+        const args = [
+            params.orders,
+            params.makerAssetFillAmount,
+            params.signatures,
+        ];
+        const calldataHexString = abiEncoder.encode(args);
+        return {
+            calldataHexString,
+            methodAbi,
+            to,
+            ethAmount,
+        };
+    }
+
+    public async getSmartContractParamsOrThrowAsync(quote: SwapQuote, opts: Partial<SwapQuoteGetOutputOpts>): Promise<SmartContractParamsInfo<ExchangeMarketBuySmartContractParams>> {
```

Comment:
I don't think you're using the `opts` here.

Reply:
Later, when supporting marketSell, the opts will probably be used.